tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -445,4 +445,177 @@ def test_consumer_groups(self):
+                parts_dirs = self._listdir(path, node, only_dirs=True)
+
+                # Merge the lists of partitions together.
+                all_parts_dirs += parts_dirs
+
+            for part_dir in all_parts_dirs:
+                path = f'{root}/{d}/kafka/{self.topic}/{part_dir}'
+                files = self._listdir(path, node)
+
+                # The data within the SI cache is a .log.N segment file
+                # where N is an positive integer
+                # Trim off the .N of the segment file
+                temp = [_file[:-2] for _file in files]
+
+                # Merge the list of files
+                segment_files += temp
+
+        for segment_file in segment_files:
+            match = re.match(r'(^(\d+)-(\d+)-([\x00-\x7F]+).log$)',
+                             segment_file)
+            assert match, f'Invalid segment filename {segment_file}'
+            assert segment_file in data_dir_segments, f'{segment_file} was not in data dir'
+
+    @cluster(num_nodes=5)
+    def test_data_loss(self):
+        # 1k messages of size 2**15
+        # is ~32MB of data.
+        msg_size = 2**15
+        msg_count = 1000
+
+        # Node alloc is first because CI fails in the debug pipeline due to
+        # the warning message "Test requested X nodes, used only Y"
+        producer = FranzGoVerifiableProducer(self.test_context, self.redpanda,
+                                             self.topic, msg_size, msg_count,
+                                             self._fgo_node)
+
+        consumer = RpkConsumer(self.test_context,
+                               self.redpanda,
+                               self.topic,
+                               num_msgs=msg_count)
+
+        self.logger.info(f"Environment: {os.environ}")
+        if os.environ.get('BUILD_TYPE', None) == 'debug':
+            self.logger.info(
+                "Skipping test in debug mode (requires release build)")
+            return
+
+        # Remote write/read and retention set at topic level
+        rpk = RpkTool(self.redpanda)
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.write', 'true')
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.read', 'true')
+        rpk.alter_topic_config(self.topic, 'retention.bytes',
+                               str(self.segment_size))
+        # Small retention.ms so we do not have to wait that long
+        # for segment removal
+        retention_ms = 120000
+        rpk.alter_topic_config(self.topic, 'retention.ms', str(retention_ms))
+
+        self._gen_manifests(msg_size, self._fgo_node)
+
+        # Produce some data
+        producer.start(clean=False)
+        producer.wait()
+
+        # Profile the data dir
+        self.logger.debug('Before retention')
+        data_dir_segments = []
+        self._find_segment_files_on_all_nodes(data_dir_segments)
+
+        # Wait for retention to kick in.
+        time.sleep((retention_ms / 1000))
```

Comment:
Rather than a straight sleep, prefer to do a wait_until for the segment count on the redpanda node to fall as a result of retention enforcement (check out other tests using `wait_for_segments_removal`)